libs/media-player/src/lib/audio-annotator.tsx

```diff
@@ -0,0 +1,274 @@
+import { isNullOrUndefined } from '@coscrad/validation-constraints';
+import {
+    ArrowLeft as ArrowLeftIcon,
+    ArrowRight as ArrowRightIcon,
+    Clear as ClearIcon,
+} from '@mui/icons-material/';
+import { Box, IconButton, Stack, Typography, styled } from '@mui/material';
+import { useRef, useState } from 'react';
+import { FormattedMediaTime } from './formatted-currenttime';
+import { KeyboardKey, useKeyDown } from './use-key-down';
+
+/**
+ * This is a duplicate of the enum in the Audio component, it should live in
+ * it's own file
+ */
+export enum AudioMIMEType {
+    mp3 = 'audio/mpeg',
+    audioMp4 = 'audio/mp4',
+    audioOgg = 'audio/ogg',
+    // TODO change this to audio/wav. This requires a migration
+    wav = 'audio/x-wav',
+    audioWebm = 'audio/webm',
+}
+
+const isAudioMIMEType = (input: unknown): input is AudioMIMEType =>
+    Object.values(AudioMIMEType).some((value) => value === (input as AudioMIMEType));
+
+const isValidTimeRangeSelection = (timeRangeSelection: TimeRangeSelection, duration: number) => {
+    const { inPointSeconds, outPointSeconds } = timeRangeSelection;
+
+    if (outPointSeconds >= duration) return false;
+
+    return outPointSeconds > inPointSeconds;
+};
+
+export type TimeRangeSelection = {
+    inPointSeconds: number;
```

Comment:
In the end I kept to seconds instead of milliseconds in the player because the native `currentTime` is in seconds (updates roughly 4 times per second).  I think it makes sense to just make the conversion to milliseconds at the interface between the web of knowledge and the media player

Reply:
Great. As long as you are explicit about units, this should work well.